src/NServiceBus.Core/Serializers/XML/Serializer.cs

```diff
@@ -372,13 +372,14 @@ void Write(StringBuilder builder, Type t, object obj)
                 return;
             }
 
-            IEnumerable<PropertyInfo> properties;
-            if (!cache.typeToProperties.TryGetValue(t, out properties))
+            Tuple<FieldInfo[], PropertyInfo[]> members;
+            if (!cache.typeMembers.TryGetValue(t, out members))
             {
-                throw new InvalidOperationException(string.Format("Type {0} was not registered in the serializer. Check that it appears in the list of configured assemblies/types to scan.", t.FullName));
+                cache.InitType(t);
```

Comment:
I think we don't need to backport the runtime initialization logic, so let's keep throwing an exception here?

Reply:
I double checked the fix. You are right. The initialization was introduced in v6 not in the backported fix. Will undo.